ci/validate.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+
+# This script is intended to be run by Cirrus-CI to validate PR
+# content prior to building any images.  It should not be run
+# under any other context.
+
+set -eo pipefail
+
+SCRIPT_FILEPATH=$(realpath "${BASH_SOURCE[0]}")
+SCRIPT_DIRPATH=$(dirname "$SCRIPT_FILEPATH")
+REPO_DIRPATH=$(realpath "$SCRIPT_DIRPATH/../")
+
+# shellcheck source=./lib.sh
+source "$REPO_DIRPATH/lib.sh"
+
+req_env_vars CIRRUS_PR CIRRUS_BASE_SHA
+
+# die() will add a reference to this file and line number.
+[[ "$CIRRUS_CI" == "true" ]] || \
+  die "This script is only/ever intended to be run by Cirrus-CI."
+
+for target in image_builder/gce.json base_images/cloud.json \
+              cache_images/cloud.json win_images/win-server-wsl.json; do
+  if ! make $target; then
+    die "Running 'make $target' failed, please validate input YAML files."
+  fi
+done
+
+# Variable is defined by Cirrus-CI at runtime
+# shellcheck disable=SC2154
+if ! git diff --name-only ${CIRRUS_BASE_SHA}..HEAD | grep -q IMG_SFX; then
+  die "Every PR must include an updated IMG_SFX file.
+Simply run 'make IMG_SFX', commit the result, and re-push."
+fi
+
+# Verify new IMG_SFX value always sorts later than previous value.  This prevents
+# screwups due to local timezone, bad, or unset clocks, etc.
```

Comment:
Oh ya, there's a setting for that: "Require branches to be up to date before merging".  I'm hesitant to turn it on now, because it will force people to constantly rebuild their images, even when doing so would be anti-helpful:  PRs in this repo. tend to take a long time to merge, and constantly rebasing them will mean constantly updating the IMG_SFX in downstream CI PR's :confounded: 

So for now, I think this might be a temporary problem until the general matter of image-update workflows is refined (and documented better).  Evenso, ultimately what will matter the most is the order IMG_SFX values are tagged in the repo.  That's the action that will trigger renovate to open image-update PRs for downstream CI.  Anyway...that's a future problem.